Doc/c-api/object.rst

```diff
@@ -6,6 +6,40 @@ Object Protocol
 ===============
 
 
+.. c:function:: PyObject* Py_GetConstant(unsigned int constant_id)
+
+   Get a :term:`strong reference` to a constant.
+
+   Return ``NULL`` if *constant_id* is invalid.
+
+   Constant identifiers:
+
+   .. c:macro: Py_CONSTANT_NONE
+   .. c:macro: Py_CONSTANT_FALSE
+   .. c:macro: Py_CONSTANT_TRUE
+   .. c:macro: Py_CONSTANT_ELLIPSIS
+   .. c:macro: Py_CONSTANT_NOT_IMPLEMENTED
+   .. c:macro: Py_CONSTANT_ZERO
+   .. c:macro: Py_CONSTANT_ONE
+   .. c:macro: Py_CONSTANT_EMPTY_STR
+   .. c:macro: Py_CONSTANT_EMPTY_BYTES
+   .. c:macro: Py_CONSTANT_EMPTY_TUPLE
+
+   .. versionadded:: 3.13
+
+   .. impl-detail::
+
+      In CPython, all of these constants are :term:`immortal`.
+
+
+.. c:function:: PyObject* Py_GetConstantBorrowed(unsigned int constant_id)
+
+   Similar to :c:func:`Py_GetConstant`, but return a :term:`borrowed
+   reference`.
+
+   The reference is borrowed from the interpreter, and will be valid until the
+   interpreter finalization.
```

Comment:
Should we add "This is primarily intended for backwards compatibility, and new users should prefer `Py_GetConstant`"?

Reply:
IMO, it's perfectly fine to use the `Borrowed` API, in this one case of interpreter-wide singletons.

(AFAIK in CPython they're even process-wide; *that* would be an implementation detail.)

Reply:
I added the sentence:

> This function is primarily intended for backwards compatibility: using :c:func:`Py_GetConstant` is recommended for new code.

IMO it's a good thing to avoid borrowed references.

Since I would like to add these functions to pythoncapi-compat which works on Python 3.6 which doesn't have immortal objects, IMO it's good to have Py_GetConstantBorrowed() **API**. Also, it should help other implementations which don't have immortal objects.